substrate/frame/offences/benchmarking/src/lib.rs

```diff
@@ -184,220 +176,7 @@ fn make_offenders<T: Config>(
-) -> Result<(Vec<pallet_im_online::IdentificationTuple<T>>, Vec<Offender<T>>), &'static str> {
-	Staking::<T>::new_session(0);
-
-	let mut offenders = vec![];
-	for i in 0..num_offenders {
-		let offender = create_offender::<T>(i + 1, num_nominators)?;
-		offenders.push(offender);
-	}
-
-	Staking::<T>::start_session(0);
-
-	let id_tuples = offenders
-		.iter()
-		.map(|offender| {
-			<
-				<T as ImOnlineConfig>::ValidatorSet as ValidatorSet<T::AccountId>
-			>::ValidatorIdOf::convert(offender.controller.clone())
-			.expect("failed to get validator id from account id")
-		})
-		.map(|validator_id| {
-			<
-				<T as ImOnlineConfig>::ValidatorSet as ValidatorSetWithIdentification<T::AccountId>
-			>::IdentificationOf::convert(validator_id.clone())
-			.map(|full_id| (validator_id, full_id))
-			.expect("failed to convert validator id to full identification")
-		})
-		.collect::<Vec<pallet_im_online::IdentificationTuple<T>>>();
-	Ok((id_tuples, offenders))
-}
-
-#[cfg(test)]
-fn check_events<
-	T: Config,
-	I: Iterator<Item = Item>,
-	Item: sp_std::borrow::Borrow<<T as SystemConfig>::RuntimeEvent> + sp_std::fmt::Debug,
->(
-	expected: I,
-) {
-	let events = System::<T>::events()
-		.into_iter()
-		.map(|frame_system::EventRecord { event, .. }| event)
-		.collect::<Vec<_>>();
-	let expected = expected.collect::<Vec<_>>();
-
-	fn pretty<D: sp_std::fmt::Debug>(header: &str, ev: &[D], offset: usize) {
-		log::info!("{}", header);
-		for (idx, ev) in ev.iter().enumerate() {
-			log::info!("\t[{:04}] {:?}", idx + offset, ev);
-		}
-	}
-	fn print_events<D: sp_std::fmt::Debug, E: sp_std::fmt::Debug>(
-		idx: usize,
-		events: &[D],
-		expected: &[E],
-	) {
-		let window = 10;
-		let start = idx.saturating_sub(window / 2);
-		let end_got = (idx + window / 2).min(events.len());
-		pretty("Got(window):", &events[start..end_got], start);
-		let end_expected = (idx + window / 2).min(expected.len());
-		pretty("Expected(window):", &expected[start..end_expected], start);
-		log::info!("---------------");
-		let start_got = events.len().saturating_sub(window);
-		pretty("Got(end):", &events[start_got..], start_got);
-		let start_expected = expected.len().saturating_sub(window);
-		pretty("Expected(end):", &expected[start_expected..], start_expected);
-	}
-
-	for (idx, (a, b)) in events.iter().zip(expected.iter()).enumerate() {
-		if a != sp_std::borrow::Borrow::borrow(b) {
-			print_events(idx, &events, &expected);
-			log::info!("Mismatch at: {}", idx);
-			log::info!("     Got: {:?}", b);
-			log::info!("Expected: {:?}", a);
-			if events.len() != expected.len() {
-				log::info!(
-					"Mismatching lengths. Got: {}, Expected: {}",
-					events.len(),
-					expected.len()
-				)
-			}
-			panic!("Mismatching events.");
-		}
-	}
-
-	if events.len() != expected.len() {
-		print_events(0, &events, &expected);
-		panic!("Mismatching lengths. Got: {}, Expected: {}", events.len(), expected.len(),)
-	}
-}
-
 benchmarks! {
-	report_offence_im_online {
```

Comment:
should we keep the benches until we remove the pallet altogether? otherwise it might be problematic to rerun benches for the fellowship runtimes unless im-online is removed from kusama and polkadot at the same time. we might want to do kusama first, so this sounds a bit problematic, although I don't see a reason why we can't do both at the same time

Reply:
I see, not introduced here, but the dependencies sound backwards. The benches for im-online pallet should live in im-online pallet, not offences. I am fine with removing them if it works and doesn't cause issues.